lib/internal/worker.js

```diff
@@ -259,6 +259,15 @@ class Worker extends EventEmitter {
                          ...new SafeArrayIterator(options.transferList));
 
     this[kPublicPort] = port1;
+    const {
+      port1: toWorkerThread,
+      port2: toHooksThread,
+    } = new MessageChannel();
+    if (!isInternal) {
+      // This is not an internal hooks thread => it needs a channel to the hooks thread:
```

Comment:
is the `isInternal` flag only used by hooks thread?
if yes, the name should be likely changed (maybe in a followup).

Reply:
Currently but not inherently limited to